barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -67,63 +83,38 @@ bool ClientIVC::verify(Proof& proof, const std::vector<VerifierAccumulator>& ver
  */
 HonkProof ClientIVC::decider_prove() const
 {
-    GoblinUltraDeciderProver decider_prover(prover_fold_output.accumulator);
+    GoblinUltraDeciderProver decider_prover(fold_output.accumulator);
     return decider_prover.construct_proof();
 }
 
 /**
- * @brief Precompute the array of verification keys by simulating folding. There will be 4 different verification keys:
- * initial function verification key (without recursive merge verifier), subsequent function verification key (with
- * recursive merge verifier), initial kernel verification key (with recursive merge verifier appended, no previous
- * kernel to fold), "full" kernel verification key( two recursive folding verifiers and merge verifier).
+ * @brief Given a set of circuits, compute the verification keys that will be required by the IVC scheme
+ * @details The verification keys computed here are in general not the same as the verification keys for the
+ * raw input circuits because recursive verifier circuits (merge and/or folding) may be appended to the incoming
+ * circuits as part accumulation.
+ * @note This method exists for convenience and is not not meant to be used in practice for IVC. Given a set of
+ * circuits, it could be run once and for all to compute then save the required VKs. It also provides a convenient
+ * (albeit innefficient) way of separating out the cost of computing VKs from a benchmark.
  *
- * TODO(https://github.com/AztecProtocol/barretenberg/issues/904): This function should ultimately be moved outside of
- * this class since it's used only for testing and benchmarking purposes and it requires us to clear state afterwards.
- * (e.g. in the Goblin object)
+ * @param circuits A copy of the circuits to be accumulated
+ * @return std::vector<std::shared_ptr<ClientIVC::VerificationKey>>
  */
-void ClientIVC::precompute_folding_verification_keys()
+std::vector<std::shared_ptr<ClientIVC::VerificationKey>> ClientIVC::precompute_folding_verification_keys(
+    std::vector<ClientCircuit> circuits)
```

Comment:
You're intentionally _not_ passing by reference here, right?

Reply:
Yes, added an extra comment to ensure this is clear